### DISCUSSION TITLE
使用Xray(M+F+B+D+G+A)+Caddy(N)+Nginx配置后无法连通reality，其他配置可以

### DISCUSSION THREAD

**Discussion by jliveusa** [lxhao61/integrated-examples#196] [Q&A] [answered]
参考了Xray(M+F+B+D+G+A)+Caddy(N)+Nginx配置，caddy使用acmesh的证书，其他配置只替换成自有域名，启动后无报错，但只有reality无法连通，不知是否证书设置出错，求指导。

**Comment by jliveusa** [comment]
使用的通配符：
  "tls": {
      "certificates": {
        "load_files": [{
          "certificate": "/home/ssl/xx.yy.crt", 
          "key": "/home/ssl/xx.yy.key", 
          "tags": ["tls01"]
        }]
换成Xray(M+F+B+D+G+A)+Caddy(N)后reality正常使用

**Comment by lxhao61** [comment]
参考的是配置 1，还是配置 2？
若采用 acmesh 获取证书，建议直接使用 caddy 申请证书即可（内置 acme，也支持申请通配符证书。）。

**Reply by jliveusa** [reply]
配置2，使用uds连接，因为有其他服务，就一直单用的acmesh

**Comment by lxhao61** [comment, marked as answer]
> 配置2，使用uds连接，因为有其他服务，就一直单用的acmesh

配置 2 注意 VLESS+Vision+REALITY 的 "dest" 为 Local Loopback 连接。
其它服务也可以由 caddy 申请及管理证书，此情况下再用 acmesh 申请及管理证书多此一举。
